src/Partials/AlignBoxButton.vala

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2019 Alecaddd (http://alecaddd.com)
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program; if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA
+ *
+ * Authored by: Giacomo "giacomoalbe" Alberini <giacomoalbe@gmail.com>
+ */
+public class Akira.Partials.AlignBoxButton : Gtk.Grid {
```

Comment:
Also, why did you make it a Grid? :) 

If this is a button, we could just have the class be a Button instead, and we can always add the icon later in the constructor. This will drop the number of grids :)

Reply:
I made it a Grid since I wanted to be prepared in the case in which we wanted a label to show right under the button. 

Even though this scenario hasn't been pictured in the mockup, I though it would have been wise enough to add it in cased we would have needed it. 

Also, HeaderBarButtons are Grids, as for the label they have. 

But I think a button would make the trick the same way, and those buttons may never see a label, due to the lack of space in that area, so, sure, I'll make the switch from Grid to Buttons :)

Reply:
Mmmm as far as I know, there isn't a real way to do it.... The whole app is under the same process.... But we could probably do it by adding a function to get the newest window from the application, and have everything that uses the state hold a reference to it. That way when a new window is created, the old widgets still hold a reference to the original window

Reply:
Yeah, let those be buttons. There's no space and it wouldn't be doable to have labels there.